bear-engine/core/src/visual/code/gl_renderer.cpp

```diff
@@ -727,7 +725,7 @@ void bear::visual::gl_renderer::copy_texture_pixels
  */
 bool bear::visual::gl_renderer::ensure_window_exists()
 {
-  boost::mutex::scoped_lock lock( m_mutex.window );
+  boost::lock(m_mutex.window, m_mutex.gl_access);
```

Comment:
I have no idea whether removing the lock/unlock in pause/unpause is correct, but this change here can't be right. If the function returns early (three lines below this) then the mutexes will not be unlocked.

This should work:

```suggestion
  boost::lock(m_mutex.window, m_mutex.gl_access);
  boost::mutex::scoped_lock window_lock( m_mutex.window, boost::adopt_lock);
  boost::mutex::scoped_lock gl_lock( m_mutex.gl_access, boost::adopt_lock);
```

This will give ownership of the locks to `scoped_lock` objects, which will automatically unlock them.